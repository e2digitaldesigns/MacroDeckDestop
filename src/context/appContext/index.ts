/* istanbul ignore file */
export * from "./appContext";
export * from "./defaultState";
