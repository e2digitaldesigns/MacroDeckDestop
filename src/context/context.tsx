import React from "react";
import { TemplateContextProvider } from "./templateContext";
import { ThemeContextProvider } from "./themeContext";

export interface IThemeComponents {
  TemplateProvider: React.ElementType;
  ThemeProvider: React.ElementType;
}

export const ApplicationContext: React.FC & IThemeComponents = () => <div />;

ApplicationContext.TemplateProvider = TemplateContextProvider;
ApplicationContext.ThemeProvider = ThemeContextProvider;
