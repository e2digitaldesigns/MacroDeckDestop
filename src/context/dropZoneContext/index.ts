export * from "./dropZoneContext";
export * from "./defaultState";
