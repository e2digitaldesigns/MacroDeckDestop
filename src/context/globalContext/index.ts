export * from "./globalContext";
export * from "./defaultState";
