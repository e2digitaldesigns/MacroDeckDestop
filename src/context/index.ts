export * from "./appContext";
export * from "./dropZoneContext";
export * from "./globalContext";
export * from "./templateContext";
export * from "./themeContext";
