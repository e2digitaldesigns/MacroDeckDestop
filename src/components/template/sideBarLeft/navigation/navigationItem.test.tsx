import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import NavigationItem from "./navigationItem";

import { AppContext } from "../../../../context/context";

const mockData = [
  { name: "Local", pads: 6 },
  { name: "Remote", pads: 12 },
  { name: "Issues", pads: 9 },
  { name: "Teams", pads: 15 },
  { name: "Tags", pads: 18 },
  { name: "Local", pads: 6 },
  { name: "Remote", pads: 12 },
  { name: "Issues", pads: 9 },
  { name: "Teams", pads: 15 },
  { name: "Tags", pads: 18 }
];

const testSetup = () => {
  return render(
    <AppContext.ThemeProvider>
      <NavigationItem data={mockData} />
    </AppContext.ThemeProvider>
  );
};

describe("<Left Navigation Item/>", () => {
  const wrapper = testSetup();
  it("Should render without errors", () => {
    const component = wrapper;
    expect(component).toMatchSnapshot();
  });
});
