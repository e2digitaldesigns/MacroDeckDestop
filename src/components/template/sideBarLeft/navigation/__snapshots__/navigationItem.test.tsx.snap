// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Left Navigation Item/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gsnTZi jhevQA"
      >
        <div>
          <svg
            fill="currentColor"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 3.5A1.5 1.5 0 0 1 2.5 2h2.764c.958 0 1.76.56 2.311 1.184C7.985 3.648 8.48 4 9 4h4.5A1.5 1.5 0 0 1 15 5.5v7a1.5 1.5 0 0 1-1.5 1.5h-11A1.5 1.5 0 0 1 1 12.5v-9zM2.5 3a.5.5 0 0 0-.5.5V6h12v-.5a.5.5 0 0 0-.5-.5H9c-.964 0-1.71-.629-2.174-1.154C6.374 3.334 5.82 3 5.264 3H2.5zM14 7H2v5.5a.5.5 0 0 0 .5.5h11a.5.5 0 0 0 .5-.5V7z"
            />
          </svg>
        </div>
        <div />
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gsnTZi jhevQA"
    >
      <div>
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 3.5A1.5 1.5 0 0 1 2.5 2h2.764c.958 0 1.76.56 2.311 1.184C7.985 3.648 8.48 4 9 4h4.5A1.5 1.5 0 0 1 15 5.5v7a1.5 1.5 0 0 1-1.5 1.5h-11A1.5 1.5 0 0 1 1 12.5v-9zM2.5 3a.5.5 0 0 0-.5.5V6h12v-.5a.5.5 0 0 0-.5-.5H9c-.964 0-1.71-.629-2.174-1.154C6.374 3.334 5.82 3 5.264 3H2.5zM14 7H2v5.5a.5.5 0 0 0 .5.5h11a.5.5 0 0 0 .5-.5V7z"
          />
        </svg>
      </div>
      <div />
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
